src/odemis/driver/tlaptmf.py

```diff
@@ -374,27 +374,27 @@ def _ReadMessage(self, timeout=None):
             for i in range(6):
                 char = self._serial.read() # empty if timeout
                 if not char:
-                    raise IOError("Controller timed out, after receiving '%s'" % msg)
+                    raise IOError("Controller timed out, after receiving '%s'" % to_str_escape(msg))
 
                 msg += char
         finally:
             self._serial.timeout = old_timeout
 
         mid = struct.unpack("<H", msg[0:2])[0]
-        if not (ord(msg[4]) & 0x80): # short message
-            logging.debug("Received: '%s'", ", ".join("%02X" % ord(c) for c in msg))
+        if not (ord(msg[4:5]) & 0x80): # short message
+            logging.debug("Received: '%s'", ", ".join("%02X" % ord(c) for c in to_str_escape(msg)))
```

Comment:
All these `join()`s are for a different output. So you need to write this way:
`", ".join("%02X" % c for c in bytearray(msg))`

You could consider putting it in a separate function, like `to_str_escape()` but that's not really needed I think.